rest/handler.go

```diff
@@ -331,7 +331,9 @@ func (h *handler) checkAuth(context *db.DatabaseContext) error {
 		h.user = context.Authenticator().AuthenticateUser(userName, password)
 		if h.user == nil {
 			base.LogfR("HTTP auth failed for username=%q", base.UD(userName))
-			h.response.Header().Set("WWW-Authenticate", `Basic realm="Couchbase Sync Gateway"`)
+			if h.server.config.SendWWWAuthenticateHeader == nil || *h.server.config.SendWWWAuthenticateHeader {
```

Comment:
I've copied this logic from elsewhere but it's basically saying:

`if not set || if set and truthy`